src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRulesResource.swift

```diff
@@ -0,0 +1,132 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import HealthCertificateToolkit
+
+struct DCCRulesResource: Resource {
+
+	// MARK: - Init
+
+	init(
+		isFake: Bool = false,
+		ruleType: HealthCertificateValidationRuleType
+	) {
+		self.locator = .DCCRules(ruleType: ruleType, isFake: isFake)
+		self.type = .caching()
+		self.sendResource = EmptySendResource()
+		self.receiveResource = CBORReceiveResource<ValidationRulesModel>()
+		self.ruleType = ruleType
+	}
+
+	// MARK: - Overrides
+
+	// MARK: - Protocol Resource
+
+	typealias Send = EmptySendResource
+	typealias Receive = CBORReceiveResource<ValidationRulesModel>
+	typealias CustomError = DCCDownloadRulesError
+
+	var locator: Locator
+	var type: ServiceType
+	var sendResource: EmptySendResource
+	var receiveResource: CBORReceiveResource<ValidationRulesModel>
+
+	func customError(for error: ServiceError<DCCDownloadRulesError>) -> DCCDownloadRulesError? {
+		switch error {
+		case .transportationError:
+			return .NO_NETWORK
+		case .unexpectedServerError(let statusCode):
+					switch statusCode {
+					case (400...499):
+						return .RULE_CLIENT_ERROR(ruleType)
+					default:
+						return .RULE_SERVER_ERROR(ruleType)
+					}
+		case let .resourceError(resourceError):
+			return handleResourceError(resourceError)
+		default:
+			return nil
+		}
+	}
+
+#if !RELEASE
+	var defaultMockLoadResource: LoadResource? = {
```

Comment:
Is it only used by tests? Can we move it to the corresponding test?

Reply:
Let me check it. If so, we would need to do it also for onboardedCountries and getTestResult, in these two we have the same pattern.